Gordon360/Services/HousingService.cs

```diff
@@ -531,4 +533,546 @@ public bool ChangeApplicationDateSubmitted(int applicationID)
+                RD_Email = ra.RD_Email,                  // RD's email
+                RD_Name = ra.RD_Name,                   // RD's name
+                RA_Profile_Link = ra.RA_Profile_Link,  // RA's profile link
+                RD_Profile_Link = ra.RD_Profile_Link, // RD's profile link
+                RA_Photo = ra.RA_Photo               // RA's Photo URL
+            })
+            .FirstOrDefaultAsync();
+
+        return onCallRA;
+    }
+
+    /// <summary>
+    /// Checks an RA in
+    /// </summary>
+    /// <param name="checkin">The viewmodel object of the RA checking in</param>
+    /// <returns>true if RA checked in successfully</returns>
+    public async Task<bool> RA_CheckinAsync(RA_On_CallViewModel checkin)
+    {
+        foreach (string hallId in checkin.Hall_ID)
+        {
+            // Check if there is an existing RA checked into this hall without an end time
+            var existingRA = await context.RA_On_Call
+                .Where(r => r.Hall_ID == hallId && r.Check_out_time == null)
+                .FirstOrDefaultAsync();
+
+            // If an existing RA is found, set their Check_out_time to the current time
+            if (existingRA != null)
+            {
+                existingRA.Check_out_time = DateTime.Now;
+                context.RA_On_Call.Update(existingRA);
+            }
+
+            // Add the new RA check-in record with no check-out time
+            var newCheckin = new RA_On_Call
+            {
+                Ra_ID = checkin.Ra_ID,
+                Hall_ID = hallId,
+                Check_in_time = DateTime.Now,
+                Check_out_time = null // RA has an active checkin
+            };
+            await context.RA_On_Call.AddAsync(newCheckin);
+        }
+
+        await context.SaveChangesAsync();
+        return true;
+    }
+
+        /// <summary>
+        /// Gets the on-call RAs for all halls.
+        /// </summary>
+        /// <returns>The RAs on call</returns>
+        public async Task<List<RA_On_Call_GetViewModel>> GetOnCallRAAllHallsAsync()
+        {
+            var onCallRAs = await context.Current_On_Call  // Use your updated view name here
+                .Select(oncall => new RA_On_Call_GetViewModel
+                {
+                    Hall_ID = oncall.Hall_ID,
+                    Hall_Name = oncall.Hall_Name,  // Hall name
+                    RA_Name = oncall.RA_Name,  // RA's full name
+                    PreferredContact = oncall.PreferredContact,  // Preferred contact method
+                    Check_in_time = oncall.Check_in_time,  // Check-in time
+                    RD_Email = oncall.RD_Email,  // RD's email
+                    RA_Profile_Link = oncall.RA_Profile_Link,  // RA's profile link
+                    RD_Profile_Link = oncall.RD_Profile_Link,  // RD's profile link
+                    RD_Name = oncall.RD_Name,
+                    RA_Photo = oncall.RA_Photo
+                })
+                .ToListAsync();
```

Comment:
Again, these look like LLM generated comments. Please remove them as they do not enhance the code.